src/ripple/protocol/impl/STObject.cpp

```diff
@@ -116,7 +116,8 @@ bool STObject::setType (const SOTemplate& type)
                 JLOG (debugJournal().warn())
                     << "setType(" << getFName().getName()
                     << "): explicit default " << e->e_field.fieldName;
-                valid = false;
+                Throw<std::runtime_error>("Explicit default " +
+                    e->e_field.fieldName);
```

Comment:
Same thing here, we took a function that uses to return a `bool` indicating success and now throws an exception based on untrusted network inputs.